### ISSUE TITLE
youtube-dl taken down - future of podsync?

### ISSUE THREAD

**Issue by richardwarriner** [mxpv/podsync#191]
https://www.bleepingcomputer.com/news/software/youtube-dl-removed-from-github-after-riaa-dmca-notice/

I guess it likely Google will make a breaking change quite soon and the current version will stop working,

Are there alternatives?

**Comment by jzvi12** [comment]
Can Streamlink be an alternative? https://github.com/streamlink/streamlink

**Comment by Rpsl** [comment]
they will migrate to another host and everything will be as well

**Comment by eduncan911** [comment]
There are many many forks.  The [Streisand effect](https://en.wikipedia.org/wiki/Streisand_effect) has hit full force with this one.

See this thread, as just one example: https://twitter.com/lrvick/status/1319729448166580227

---

Tip: As per the Streisand effect, I found https://github.com/blackjack4494/yt-dlc that is an updated fork, since youtube-dl hasn't been maintained in years.

**Comment by mxpv** [comment]
Waiting for things to settle down a bit. Alternative backend is also an option (especially written in Go). Going to research this.

**Comment by jzvi12** [comment]
I'm seeing the following from the logs:

```
ERRO[2020-11-01T05:03:33Z] youtube-dl error: /tmp/podsync-500258317/XxXxxxXXxx.%(ext)s  error="failed to execute youtube-dl: signal: killed"
...
ERRO[2020-11-01T05:13:44Z] youtube-dl error: /tmp/podsync-895950279/XXXxxxxXxxX.%(ext)s  error="failed to execute youtube-dl: exit status 1"
```

Would this be an issue related to youtube-dl being outdated and Youtube making changes on their side?

my yt-dl version is:

```
$ youtube-dl --version
2020.05.29
```

**Comment by richardwarriner** [comment]
https://www.bleepingcomputer.com/news/software/github-threatens-to-ban-users-who-bypass-youtube-dl-takedown/

**Comment by jtagcat** [comment]
Are there any other examples on how repos with the shared dependency have handled, or are planning to handle it?

**Comment by jtagcat** [comment]
> https://www.bleepingcomputer.com/news/software/github-threatens-to-ban-users-who-bypass-youtube-dl-takedown/

Thus switching to a different source is probably not in scope. The readme says to install it from brew.

I think of incident response, we are talking about the docker image.
A workaround could be to install it from the package manager, as the latest (`2020.11.01.1`) probably won't change so often for now. Getting it from the package manager, and the developers pushing to there from a different source is out of scope of our problem, and I'd consider it the best option.

As for changing it to a non-github mirror, I think we'll rather reserve it for the possible future.

tldr:
1. we are talking about docker here
1. package manager probably has latest (`2020.11.01.1` on localhost), and it won't likely change for the period of change
1. package manager is better, thinking legal, than non-github mirror
1. if package manager can't satisfy, then probably we can't either, we'll have to wait out the shitstorm, and act accordingly

**Comment by jtagcat** [comment]
I'll cook up a MR for the proposed first incident response.

**Comment by jtagcat** [comment]
The Dockerfile assumes there is an executable, podsync at the root of the repo, but there isn't. :/

**Comment by jtagcat** [comment]
todo: add some docs on hey you can use make
hey this is my first _developer_ usage of make
podsync is now in bin/podsync instead

**Comment by jzvi12** [comment]
`time="2020-11-10T18:34:14Z" level=error msg="[youtube] XxxXXxxXXX: Downloading webpage\nERROR: XxxXXxxXXX: YouTube said: Unable to extract video data\n"`

**Comment by jtagcat** [comment]
(I would re-open and pin, @mxpv )

**Comment by eduncan911** [comment]
FYI, the repo has been restored back to github.

https://www.reddit.com/r/DataHoarder/comments/jv7pkb/youtubedls_repository_has_been_restored/
